src/observability.py

```diff
@@ -36,3 +40,49 @@ def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
         self._grafana_dashboards = GrafanaDashboardProvider(
             charm, relation_name="grafana-dashboard"
         )
+        self._charm.framework.observe(
+            self._charm.on.statsd_prometheus_exporter_pebble_ready,
+            self._on_statsd_prometheus_exporter_pebble_ready,
+        )
+
+    def _on_statsd_prometheus_exporter_pebble_ready(self, _event: ops.PebbleReadyEvent) -> None:
+        """Handle the statsd-prometheus-exporter-pebble-ready event."""
+        container = self._charm.unit.get_container("statsd-prometheus-exporter")
+        container.push(
+            "/statsd.conf",
+            textwrap.dedent(
+                """\
+                mappings:
```

Comment:
I had a review [here](https://github.com/canonical/synapse-operator/pull/30#discussion_r1315856316) regarding harded coded configuration. Do you think it applies to this case too?

Reply:
I believe this is slightly different. It's a static string that is written to the configuration file without any modifications.